srml/balances/src/lib.rs

```diff
@@ -1151,3 +1133,55 @@ where
 		Self::total_balance(who).is_zero()
 	}
 }
+
+impl<T: Trait<I>, I: Instance> Module<T, I> {
```

Comment:
Why not in `impl Currency`?https://github.com/paritytech/substrate/blob/91e73bd9609241a5c7a14a6af9bfe6677a494dc3/srml/balances/src/lib.rs#L789

Reply:
I don't want to hassle `Currency` implementors with writing an implementation of `transfer_inner`/`transfer_some`.

Reply:
we could change transfer signature of Currency trait to:
```rust
	fn transfer(transactor: &T::AccountId, dest: &T::AccountId, value: Self::Balance, liveness: ExistenceRequirement) -> Result {
```

Reply:
I think that we should try to avoid breaking anything if we can. See @xlc's first comment in this PR.

Reply:
Xlc comment is about not changing the Call. We can still have transfer and transfer_keep_alive in Call and have `transfer(...., liveness: ExistenceRequirement)` in Currency trait.

I feel generalizing the trait here doesn't harm

Reply:
Depends if anyone else has implemented anything on it - third parties might have done. That said, if it is to be changed, now is the time to do it.